src/scores/continuous/threshold_weighted_impl.py

```diff
@@ -0,0 +1,739 @@
+            of `reduce_dims` and `preserve_dims` can be supplied. The default behaviour
+            if neither are supplied is to reduce all dims.
+        preserve_dims: Optionally specify which dimensions to preserve when calculating
+            the threshold_weighted_expectile_score. All other dimensions will be reduced. As a special case, 'all'
+            will allow all dimensions to be preserved. In this case, the result will be in
+            the same shape/dimensionality as the forecast, and the errors will be the threshold_weighted_expectile_score
+            at each point (i.e. single-value comparison against observed), and the
+            forecast and observed dimensions must match precisely. Only one of `reduce_dims`
+            and `preserve_dims` can be supplied. The default behaviour if neither are supplied
+            is to reduce all dims.
+        weights: Optionally provide an array for weighted averaging (e.g. by area, by latitude,
+            by population, custom)
+
+    Returns:
+        xarray data array of the threshold weighted expectile error
+
+    Raises:
+        ValueError: if `interval_where_one` is not length 2.
+        ValueError: if `interval_where_one` is not increasing.
+        ValueError: if `alpha` is not in the open interval (0,1).
+        ValueError: if `interval_where_one` and `interval_where_positive` do not
+            specify a valid trapezoidal weight.
+
+    References:
+        Taggart, R. (2022). Evaluation of point forecasts for extreme events using
+        consistent scoring functions. Quarterly Journal of the Royal Meteorological
+        Society, 148(742), 306-320. https://doi.org/10.1002/qj.4206
+    """
+
+    check_alpha(alpha)
+    _check_tws_args(interval_where_one=interval_where_one, interval_where_positive=interval_where_positive)
+    _, phi, phi_prime = _auxiliary_funcs(fcst, obs, interval_where_one, interval_where_positive)
+    # We multiply the output by a factor of two here due to the scaling of phi and phi_prime
+    # Since phi(s)=2s^2 was used in `_auxiliary_funcs` to be consistent with Taggart (2022)
+    return 0.5 * consistent_expectile_score(
+        fcst, obs, alpha, phi, phi_prime, reduce_dims=reduce_dims, preserve_dims=preserve_dims, weights=weights
+    )
+
+
+def tw_huber_loss(
+    fcst: xr.DataArray,
+    obs: xr.DataArray,
+    huber_param: float,
+    interval_where_one: Tuple[EndpointType, EndpointType],
+    *,
+    interval_where_positive: Optional[Tuple[EndpointType, EndpointType]] = None,
+    reduce_dims: Optional[FlexibleDimensionTypes] = None,
+    preserve_dims: Optional[FlexibleDimensionTypes] = None,
+    weights: Optional[xr.DataArray] = None,
+) -> xr.DataArray:
+    """
+    Returns the threshold weighted huber loss.
+
+    For more flexible weighting schemes,
+    see :py:func:`scores.continuous.consistent_huber_score`.
+
+    Two types of threshold weighting are supported: rectangular and trapezoidal.
+
```

Comment:
Please remove the blank line on line 664. This helps with the sphinx rendering of the API docs in this situation (the whens and whys of blank lined, indentation and bullet lists are very nuanced)